node/node_syncing.go

```diff
@@ -224,32 +220,18 @@ func (node *Node) doBeaconSyncing() {
 		// If Downloader is not working, we need also deal with blocks from beaconBlockChannel
 		go func(node *Node) {
 			// TODO ek – infinite loop; add shutdown/cleanup logic
-			for beaconBlock := range node.BeaconBlockChannel {
-				if node.beaconSync != nil {
-					if beaconBlock.NumberU64() >= node.Beaconchain().CurrentBlock().NumberU64()+1 {
-						// have crosslinks broadcasted by only leader or 1% of validators to avoid p2p spamming
-						if node.Consensus.IsLeader() || rand.Intn(100) == 0 {
-							node.BroadcastCrossLinkFromShardsToBeacon()
-						}
-						err := node.beaconSync.UpdateBlockAndStatus(
-							beaconBlock, node.Beaconchain(), true,
-						)
-						if err != nil {
-							node.beaconSync.AddLastMileBlock(beaconBlock)
-						}
-					}
-				}
+			for _ = range node.BeaconBlockChannel {
```

Comment:
Why is this loop empty now?

Reply:
everything inside this loop is moved to another place or removed as useless. But we still need to read from channel, or it will block. I will consider removing this channel, but not necessarily in the current pr.